Cargo.toml

```diff
@@ -15,6 +15,7 @@ build = "build.rs"
 [dependencies]
 regex = "0.1"
 lazy_static = "0.1.15"
+itertools = "0.4.15"
```

Comment:
This is what implements the `join()` on line 296.
As I said, I wasn't sure of a better way to do String joins.  Most every reference to the question led me to this crate...